### ISSUE TITLE
"What drugs may treat drug-induced liver injury?" returns "treats" edges that are really "causes" edges

### ISSUE THREAD

**Issue by karafecho** [NCATSTranslator/Feedback#163]
This issue is to report that a query for "What drugs may treat drug-induced liver injury?" returns "treats" edges that are really "causes" edges, with evidence to refute the use of the "treats" predicate. This is a known issue, but I am reporting it again here because it seemed like an extreme example, perhaps because the disease was "**drug**-induced liver injury".

PK=2ed6f2ba-d96f-4f9e-a2a9-093ff5ea8d69

Specifically, I glanced at the first five results returned (additional answers were loading) and saw that four of five were all causal agents (acetaminophen, valproic acid, leflunomide, ketoconazole), not treatments, for DILI. A deeper dive showed that of the available evidence for the four incorrect results, three edges were from SemMedDB and two were from Text Mining KP. The correct "treats" edge for sapropel was from SemMedDB. The five results in question here appear to be from BTE.

After loading additional results, I continued to see a number of causal agents, although the "ameliorates" results looked pretty interesting. For example: 

![image](https://user-images.githubusercontent.com/26254388/234404200-fe9bbaf6-aa9f-43f8-9f65-22d507bda946.png)

![image](https://user-images.githubusercontent.com/26254388/234404440-05c7a5a6-e3b4-4e47-9e72-05b68ca1566a.png)

**Comment by sierra-moxon** [comment]
@sstemann - thanks so much, it along with its "opposite" companions are in a shared swim lane too.  I figured out that I don't have permission to add labels to tickets in the O&O repo.  Would you be able to add "architecture" to that repo as well?  (then we can add this one: https://github.com/NCATSTranslator/Ordering-Organizing/issues/21)

**Comment by cbizon** [comment]
This looks like a semmed and text-mining issue.  A quick perusal shows that I think all of the bad edges are coming from semmed, with one of them also coming from text-mining.

@andrewsu do you have any sense of whether these would be culled by your new denoising?
@bill-baumgartner is there any TM activity to take here?

**Comment by andrewsu** [comment]
Here is my semmeddb analysis:

Ketoconazole -- https://biothings.ncats.io/semmeddb2/query?q=subject.umls:C0022625%20AND%20object.umls:C0019193
* 3 PMIDs for CAUSES
* 1 PMID for TREATS

leflunomide -- https://biothings.ncats.io/semmeddb2/query?q=subject.umls:C0063041%20AND%20object.umls:C0019193
* 1 PMID for TREATS

valproic acid -- https://biothings.ncats.io/semmeddb2/query?q=(subject.umls:C0080356%20OR%20subject.umls:C0042291)%20AND%20object.umls:C0019193
* 1 PMID for TREATS

acetaminophen -- https://biothings.ncats.io/semmeddb2/query?q=(subject.umls:C0000970%20OR%20subject.umls:C1258986)%20AND%20object.umls:C0019193
* 13 PMIDs for CAUSES

sapropel -- https://biothings.ncats.io/semmeddb2/query?q=subject.umls:C0084461%20AND%20object.umls:C0019193
* 1 PMID for TREATS (correct!)

**Bottom line**: Filtering by `pmid_count > 1` would have eliminated all of the incorrect `TREATS` relations, but also would have removed the correct `sapropel - treats - DILI` result.  There don't appear to be any NER errors or Biolink domain/range constraint errors, so none of the other filtering options we're considering would have changed these results.

**Comment by bill-baumgartner** [comment]
I've reviewed all of the TMKP responses for this query. There are some difficult cases, but also many that are clearly not `treats` assertions (some of which are in fact `causes` assertions as was pointed out above). Thanks for pointing these out. We will further analyze these errors and use these examples as feedback to hopefully improve the underlying model that is identifying chemical-treats/causes-disease assertions.

**Comment by sierra-moxon** [comment]
in retesting ci today, none of the following show up: `acetaminophen`, `valproic acid`, `leflunomide`, `ketoconazole`
neither does `sapropel`

I waited for 5 mins in retesting - @karafecho - is this closeable?

**Comment by karafecho** [comment]
I also retested (I waited about 10 min) and do not see the five examples above. However, sapropel was correct, so I should be seeing that answer. I'm also seeing a few other incorrect answers, e.g., amoxicillin and clavulanic acid, that I didn't see the first time I ran the query. 

It looks like Andrew and Bill have reviewed/analyzed this example, so you can probably close it, but it may be good to retest after a SemMedDB fix is in place.

**Comment by andrewsu** [comment]
I'm going to close this ticket since the main issue here of removing the false positive examples will be addressed by https://github.com/NCATSTranslator/Feedback/issues/100. I think addressing the subsequent false negative of "sapropel" should be addressed in a different ticket, if that's something we want to follow up on.
